include/rocksdb/c.h

```diff
@@ -1252,6 +1254,10 @@ rocksdb_options_set_max_bytes_for_level_multiplier_additional(
     rocksdb_options_t*, int* level_values, size_t num_levels);
 extern ROCKSDB_LIBRARY_API void rocksdb_options_enable_statistics(
     rocksdb_options_t*);
+extern ROCKSDB_LIBRARY_API void rocksdb_options_set_statistics_level(
+    rocksdb_options_t*, int level);
```

Comment:
It's actually consistent with the way it's done in other parts of the C API. See, for example, rocksdb_options_set_compression. I'll add human readable constants for levels, though

Reply:
Can you also initialize them with int value just like 
```
enum {
  rocksdb_no_compression = 0,
  rocksdb_snappy_compression = 1,
  rocksdb_zlib_compression = 2,
  rocksdb_bz2_compression = 3,
  rocksdb_lz4_compression = 4,
  rocksdb_lz4hc_compression = 5,
  rocksdb_xpress_compression = 6,
  rocksdb_zstd_compression = 7
};
```


Also, since this enum is in c.h now, I believe we can replace the hard-coded int in your new tests that are supposed to mean stats level with enum you defined (include but not limited to https://github.com/facebook/rocksdb/pull/11263#discussion_r1247511494)

Reply:
Thanks!